src/hdf5_init_meta.c

```diff
@@ -185,7 +193,8 @@ void create_meta_mq1_fields_dataset(hid_t file, hid_t lcpl, hid_t *meta_handle)
   }
 
 cleanup:
-  // H5Tclose(header_id_type);
+  if (H5Iis_valid(header_id_type))
+    H5Tclose(header_id_type);
```

Comment:
With the current memory clean-up, the order doesn't matter, but I think you have followed a last-allocated-first-free order, for consistency, this could be placed after `pixel_depth_type`.